resources/views/visit-transfer/admin/application/view.blade.php

```diff
@@ -545,7 +559,47 @@ class="text-danger">This action cannot be undone</strong>.
                     </div>
                     <div class="modal-footer">
                         <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
-                        <button type="submit" class="btn btn-success">Accept Application - this cannot be undone</button>
+                        <button type="submit" class="btn btn-success">Accept Application - this cannot be undone
+                        </button>
+                    </div>
+                </div>
+            </div>
+            {!! Form::close() !!}
+        </div>
+
+        <div class="modal fade" id="modalApplicationComplete" tabindex="-1" role="dialog"
+             aria-labelledby="Accept Application" aria-hidden="true">
+            {!! Form::open(array("url" => URL::route("visiting.admin.application.complete.post", $application->id))) !!}
+            <div class="modal-dialog">
+                <div class="modal-content">
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span
+                                    aria-hidden="true">&times;</span></button>
+                        <h4 class="modal-title" id="myModalLabel">
+                            Complete Application #{{ $application->public_id }}
+                            - {{ $application->type_string }} {{ $application->facility_name }}
+                        </h4>
+                    </div>
+                    <div class="modal-body">
+                        <p>
+                            It is important that you understand <strong class="text-danger">this action cannot be
+                                undone.</strong>
+                        </p>
+
+                        <p>
+                            You must write a staff note detailing why you have completed the application.
```

Comment:
You say here that you must write a staff note, however you allow the note to be nullable in the validation

Reply:
Now ensuring that it is mandatory